compiler/src/dmd/errors.d

```diff
@@ -24,6 +25,57 @@ import dmd.console;
 
 nothrow:
 
+/***************************
+ * Error message sink for D compiler.
+ */
+class ErrorSinkCompiler : ErrorSink
```

Comment:
I don't think that's the right abstraction either. What about messages issued by `pragma(msg)` ?

Reply:
```
bool pragmaMsgSemantic(Loc loc, Scope* sc, Expressions* args)
{
    if (!args)
        return true;
    foreach (arg; *args)
    {
        sc = sc.startCTFE();
        auto e = arg.expressionSemantic(sc);
        e = resolveProperties(sc, e);
        sc = sc.endCTFE();

        // pragma(msg) is allowed to contain types as well as expressions
        e = ctfeInterpretForPragmaMsg(e);
        if (e.op == EXP.error)
        {
            errorSupplemental(loc, "while evaluating `pragma(msg, %s)`", arg.toChars());
            return false;
        }
        if (auto se = e.toStringExp())
        {
            const slice = se.toUTF8(sc).peekString();
            fprintf(stderr, "%.*s", cast(int)slice.length, slice.ptr);
        }
        else
            fprintf(stderr, "%s", e.toChars());
    }
    fprintf(stderr, "\n");
    return true;
}
```
impl should really be using `message`, but anyway motivates the need for another category of output in this type of interface.